### PR TITLE
use workspace inheritance for common version

### PR DESCRIPTION
**Pull request by dead10ck**
By using the top-level `Cargo.toml` and workspace inheritance, we can eliminate all the duplication of the version number, and make `cargo build` report the correct version (rather than the old 0.6.0). This does require adding a patch version to make it semver compatible.

### CODE REVIEW

**Review by pascalkuthe — APPROVED** [review]
I would like to see the same thing being done for dependencies too (both so we get a better overview of all external dependencies and avoid duplicate versions)

File: helix-loader/build.rs [hunk 1]
```diff
@@ -2,7 +2,7 @@ use std::borrow::Cow;
 use std::path::Path;
 use std::process::Command;
 
-const VERSION: &str = include_str!("../VERSION");
```
[the remark this answers is not in the document]
Comment (dead10ck) [reply helix-loader/build.rs:5]: Yep, good idea. I updated the wording, and deleted the `VERSION` file.

**Review by the-mikedavis — APPROVED** [review]

File: docs/releases.md [hunk 2]
```diff
@@ -1,16 +1,18 @@
 ## Checklist
 
 Helix releases are versioned in the Calendar Versioning scheme:
-`YY.0M(.MICRO)`, for example, `22.05` for May of 2022. In these instructions
+`YY.MM.MICRO`, for example, `22.05.0` for May of 2022. In these instructions
```
Comment (archseer) [inline docs/releases.md:4]: I personally don't like this notational change and would prefer we keep the original format.

Note that `MM` should also be changed back to `0M` regardless 

MM - Short month - 1, 2 ... 11, 12
0M - Zero-padded month - 01, 02 ... 11, 12

https://calver.org/#scheme

Comment (dead10ck) [reply docs/releases.md:4]: >Note that MM should also be changed back to 0M regardless

Ah sorry, I was not familiar with CalVer notation until now. I had assumed it was a typo.

>I personally don't like this notational change and would prefer we keep the original format.

So just to clarify, are you talking about just the 0M thing, or having a patch version at all? Because Cargo throws an error without one.

File: Cargo.toml [hunk 3]
```diff
@@ -40,6 +40,11 @@ tree-sitter = { version = "0.20", git = "https://github.com/tree-sitter/tree-sit
 nucleo = "0.2.0"
 
 [workspace.package]
+version = "23.10.0"
```
[the remark this answers is not in the document]
Comment (dead10ck) [reply Cargo.toml:43]: Yes, it throws an error if the version is not semver. But it looks like having a patch version is a thing in CalVer, so it can be compatible with both.

**Comment by dead10ck** [comment]
Ok this was changed to still use the `Cargo.toml`, but it drops the micro version if it's equal to 0. This will let us track the project version with Cargo, while the user facing version format remains unchanged.

File: docs/releases.md [hunk 4]
```diff
@@ -1,16 +1,23 @@
 ## Checklist
 
 Helix releases are versioned in the Calendar Versioning scheme:
-`YY.0M(.MICRO)`, for example, `22.05` for May of 2022. In these instructions
-we'll use `<tag>` as a placeholder for the tag being published.
+`YY.0M(.MICRO)`, for example, `22.05` for May of 2022, or in a patch release,
```
Comment (pascalkuthe) [inline docs/releases.md:4]: Is this change intentional?

Comment (dead10ck) [reply docs/releases.md:4]: Yes. Is there an issue with it?

**Review by pascalkuthe — APPROVED** [review]
